webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,231 @@
+
+const STORAGE_PATH = 'notebook-launch';
+const STORAGE_KEY = 'notebook-config';
+
+interface Props extends ModalProps {
+  onLaunch?: () => void;
+  visible?: boolean;
+}
+
+const NotebookModal: React.FC<Props> = (
+  { visible = false, onLaunch, ...props }: Props,
+) => {
+  const storage = useStorage(STORAGE_PATH);
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ showResourceType, setShowResourceType ] = useState(true);
+  const [ resourceType, setResourceType ] = useState<ResourceType | undefined>
+  (storage.getWithDefault(STORAGE_KEY, { type: undefined }).type);
+  const [ form ] = Form.useForm();
+
+  const fetchTemplates = useCallback(async () => {
+    try {
+      setTemplates(await getTaskTemplates({}));
+    } catch {}
+  },[]);
+
+  useEffect(() => {
+    fetchTemplates();
+  }, [ fetchTemplates ]);
+
+  const fetchResourcePools = useCallback(async () => {
+    try {
+      setResourcePools( await getResourcePools({}));
+    } catch {}
+  }, []);
+
+  useEffect(() => {
+    fetchResourcePools();
+  }, [ fetchResourcePools ]);
+
+  const fetchConfig = useCallback(async () => {
+    try {
+      const values: NotebookConfig = form.getFieldsValue(true);
+      const config = await previewNotebook(
+        values.type === ResourceType.CPU ? 0 : values.slots,
+        values.template === 'default'? undefined : values.template,
+        values.name,
+        values.pool,
+      );
+      form.setFieldsValue({ config: yaml.dump(config) });
+    } catch {}
+  }, [ form ]);
+
+  useEffect(()=> {
+    if (showFullConfig){
+      fetchConfig();
+    }
+  }, [ showFullConfig, fetchConfig ]);
+
+  const storeConfig = useCallback((_, values: NotebookConfig) => {
+    delete values.name;
+    storage.set(STORAGE_KEY,values);
+  }, [ storage ]);
+
+  const handleSecondary = useCallback(() => {
+    if (showFullConfig) {
+      setShowFullConfig(show => !show);
+    } else {
+      form.validateFields().then(() => setShowFullConfig(show => !show)).catch();
+    }
+  },[ form, showFullConfig ]);
+
+  const handleCreateEnvironment = useCallback(
+    (values) =>{
+      if (showFullConfig) {
+        launchNotebook(yaml.load(form.getFieldValue('config')) as RawJson);
+      } else {
+        launchNotebook(
+          undefined,
+          resourceType === ResourceType.CPU ? 0 : values.slots,
+          values.template === 'default' ? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      if (onLaunch) onLaunch();
+    },
+    [ showFullConfig, form, onLaunch, resourceType ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((e: string) => {
+    const pool = resourcePools.find(pool => pool.name === e);
+    if (pool){
+      if (pool.cpuContainerCapacityPerAgent > 0 && pool.slotsPerAgent && pool.slotsPerAgent > 0) {
```

Comment:
**suggestion:** I checked the default resource pool and saw that `slotsPerAgent` can be set to `-1` to indicate that there is GPU capacity but just no ceiling. So we may have to do the check something more like:
```
const hasCPUCapacity = selectedPool.cpuContainerCapacityPerAgent !== 0;
const hasGPUCapacity = selectedPool.slotsPerAgent !== 0;
```

Reply:
@hkang1 we should check if the value be larger than 0 rather than `!== 0`. `slotsPerAgent == -1` now means we don't have dynamic agents configured but there still might be static agents that have GPU slots.